File: 0647-palindromic-substrings/0647-palindromic-substrings.java
class Solution {
    int count ;
    public int countSubstrings(String s) {
        int n = s.length();
        for(int i = 0 ; i<n ; i++){
            for(int j = i ; j<n ; j++){
                if (f1(s.substring(i,j+1))) count++;
            }
        }
        return count ;
    }
    private boolean f1(String s ){
        if(s.length()==0) return false ;
        int n = s.length();
        for(int i = 0 ; i<n/2 ; i++){
            if(s.charAt(i)!=s.charAt(n-1-i)) return false ;
        }
        return true ;
    }
}

File: 0647-palindromic-substrings/0647-palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s)
        if(n<2):
            return n
        def palin(s):
            return s == s[::-1]
        count = 0
        for i in range(n):
            for j in range(i+1,n+1):
                t = s[i:j]
                if palin(t):
                    count+=1
        return count